src/System.Text.Json/src/System/Text/Json/Utf8JsonReader.cs

```diff
@@ -1697,6 +1696,15 @@ private bool SkipSingleLineComment(ReadOnlySpan<byte> localBuffer, out int idx)
                 return false;
             }
 
+            // If we have encountered \r, check and advance index if next character is \n
+            if (localBuffer[idx] == JsonConstants.CarriageReturn 
```

Comment:
How can we deal with an input like this that is split between the `\r` and `\n`?
```text
firstPart: "{\"name\":\"value\" \\ A comment \r
secondPart: \n}"
```

```C#
JsonReaderState state = ... // AllowComments.
var reader = new JsonReader(input.Slice(0, split), isFinalBlock: false, state)
// reader loop
// What is the comment value?
// Currently it is \\ A comment \r, which is incorrect.
// We expect the comment value to be: \\ A comment \r\n
reader = new JsonReader(input.Slice(reader.BytesConsumed), isFinalBlock: true, state)
// reader loop
// Only now we can see the end of comment and know what the comment value should be.
```

Reply:
Apologies, if I appear to be intruding in this discussion. I was thinking of working on #33321 which got me here and then had curiosity questions.

As for the example in the above comment, I was wondering if we really need to specially handle `\n` (after the `\r` in this particular case. As example, consider a valid JSON as below
```json
{ "name":"value"

}
```
with its buffer representation being `{\"name\":\"value\"\n\n}`. For such inputs, if I understand correctly, the `\n`s would be consumed by method [SkipWhiteSpace](https://github.com/dotnet/corefx/blob/0354aa6aeeb9bb50ce97ef69944cf3371496ec74/src/System.Text.Json/src/System/Text/Json/Utf8JsonReader.cs#L477)? Will the residual unhandled `\n` in `\r\n` not be consumed in similar fashion?
In other words, through the implementation point of view would it be sufficient to consider that a single line comment ends in `\r` or `\n`?

Reply:
@ahsonkhan Please refer to above :-) Again, sorry for intruding in this PR. But I am asking this out of interest of understanding how we 

Essentially, as per the standard, if the comment can be terminated with a `\r` or a `\n`, then ideally whatever follows either of these line terminator tokens could be considered as part of next line? So even if a `\n` exists after `\r` i.e. in a `\r\n`, its presence is insignificant if the parsing is considered to be _greedy_ then `\n` could well be left out as a normal `\n` which could occur anywhere as I mentioned in comment above? In fact, as far as the standard is concerned, isn't `\r\n` irrelevant given the fact that it is a combination of two characters for which their significance has already been defined.

Reply:
> Again, sorry for intruding in this PR. 

No problem.

> Essentially, as per the standard, if the comment can be terminated with a `\r` or a `\n`, then ideally whatever follows either of these line terminator tokens could be considered as part of next line?

You are correct. I was considering only `\n` or `\r\n` together implying an end of comment, not just `\r`. However, given `\r` on its own can be used as an "end of comment" indicator (following Json.NET), we don't need to explicitly check for a `\n` following it. We should add test cases to capture this, nonetheless. Essentially, we need to verify that the behavior of reading JSON containing a comment ending in any of `\n`, `\r`, and `\r\n` is identical, regardless of what follows after it (whether it is extra white space or another JSON token).